pkg/urcrypt/urcrypt/pedersen.c

```diff
@@ -0,0 +1,356 @@
+#include "urcrypt.h"
+#include "util.h"
+#include <string.h>
+#include <libec.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define MIN(a,b)                                \
+  ({ __typeof__ (a) _a = (a);                   \
+    __typeof__ (b) _b = (b);                    \
+    _a < _b ? _a : _b; })
+
+typedef struct {
+  prj_pt p0;
+  prj_pt p1;
+  prj_pt p2;
+  prj_pt p3;
+  prj_pt p4;
+} constant_points;
+
+struct urcrypt_pedersen_context_struct {
+  constant_points *cp;
+  ec_params *curve_params;
+};
+
+size_t
+urcrypt_pedersen_context_size()
```

Comment:
this whole size() concept is unnecessary for pederson -- secp uses it because the size is dynamic. This is a struct with a constant size, so you can do the whole thing with no malloc at all with the sizeof operator.